### PR TITLE
DT-1257 remove sql files and convert to kotlin enums 

### PR DESCRIPTION
**Pull request by tomsmyers**
as carefully as i made these changes, this PR scares the life out of me. it would be great if there was a way to verify that the content of the queries is unchanged. 🙏🏼

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/RepositoryBase.java [hunk 1]
```diff
@@ -20,9 +19,6 @@
     @Autowired
     protected NamedParameterJdbcOperations jdbcTemplate;
 
-    @Autowired
-    protected SQLProvider sqlProvider;
```
Comment (tomsmyers) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/RepositoryBase.java:24]: this is no longer needed. i also removed the bean itself.

File: src/main/java/uk/gov/justice/hmpps/prison/repository/StaffRepository.java [hunk 2]
```diff
@@ -24,9 +25,6 @@
 @Repository
 @Slf4j
 public class StaffRepository extends RepositoryBase {
-    private static final String NAME_FILTER_QUERY_TEMPLATE = " AND (UPPER(FIRST_NAME) LIKE '%s%%' OR UPPER(LAST_NAME) LIKE '%s%%')";
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/StaffRepository.java:27]: so what did you do about these?

File: src/main/java/uk/gov/justice/hmpps/prison/util/SQLProvider.java [hunk 3]
```diff
@@ -1,153 +0,0 @@
...
-
-    private String makeString(final char[] content, final int startIndex, final int endIndex) {
-        final var sb = new StringBuilder();
-        for (var i = startIndex; i < content.length && i < endIndex; i++) {
-            sb.append(content[i]);
-        }
-        return sb.toString();
-    }
-
-    private void parse(final char[] content) throws ParseException {
```
Comment (tomsmyers) [inline src/main/java/uk/gov/justice/hmpps/prison/util/SQLProvider.java:93]: i noticed that the old sql files get run through this when loaded

can we be confident that the copied queries are the same, now they are being read directly?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/repository/StaffRepository.java (hunk 2 above)
Comment (tomsmyers) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/StaffRepository.java:27]: any contstants that contained sql were moved to the relevant enum class. this one is now `StaffRepositorySql.NAME_FILTER_QUERY_TEMPLATE`

File: src/main/java/uk/gov/justice/hmpps/prison/util/SQLProvider.java (hunk 3 above)
Comment (tomsmyers) [reply src/main/java/uk/gov/justice/hmpps/prison/util/SQLProvider.java:93]: @andymarke @Mjwillis any thoughts on this?

**Review by Mjwillis — APPROVED** [review]
LGTM!

**Pull request merged**